src/main/java/ch/njol/skript/effects/EffLog.java

```diff
@@ -77,42 +81,61 @@ public void close() {
 	private Expression<String> messages;
 	@Nullable
 	private Expression<String> files;
-	
+
+	private Level logLevel = Level.INFO;
+	private static final int WARNING_LOG = 900;
+	private static final int SEVERE_LOG = 1000;
+
 	@SuppressWarnings({"unchecked", "null"})
 	@Override
-	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parser) {
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parser) {
 		messages = (Expression<String>) exprs[0];
 		files = (Expression<String>) exprs[1];
+		if (parser.mark == 1) {
+			logLevel = Level.WARNING;
+		} else if (parser.mark == 2) {
+			logLevel = Level.SEVERE;
+		}
 		return true;
 	}
 	
 	@SuppressWarnings("resource")
 	@Override
-	protected void execute(final Event e) {
-		for (final String message : messages.getArray(e)) {
+	protected void execute(Event event) {
+		for (String message : messages.getArray(event)) {
 			if (files != null) {
-				for (String s : files.getArray(e)) {
-					s = s.toLowerCase(Locale.ENGLISH);
-					if (!s.endsWith(".log"))
-						s += ".log";
-					if (s.equals("server.log")) {
-						SkriptLogger.LOGGER.log(Level.INFO, message);
-						continue;
+				for (String logFile : files.getArray(event)) {
+					logFile = logFile.toLowerCase(Locale.ENGLISH);
+					if (!logFile.endsWith(".log"))
+						logFile += ".log";
+					if (logFile.equals("server.log")) {
+						SkriptLogger.LOGGER.log(logLevel, message);
 					}
-					PrintWriter w = writers.get(s);
-					if (w == null) {
-						final File f = new File(logsFolder, s); // REMIND what if s contains '..'?
+					PrintWriter logWriter = writers.get(logFile);
+					if (logWriter == null) {
+						File logFolder = new File(logsFolder, logFile); // REMIND what if logFile contains '..'?
 						try {
-							f.getParentFile().mkdirs();
-							w = new PrintWriter(new BufferedWriter(new FileWriter(f, true)));
-							writers.put(s, w);
-						} catch (final IOException ex) {
-							Skript.error("Cannot write to log file '" + s + "' (" + f.getPath() + "): " + ExceptionUtils.toString(ex));
+							logFolder.getParentFile().mkdirs();
+							logWriter = new PrintWriter(new BufferedWriter(new FileWriter(logFolder, true)));
+							writers.put(logFile, logWriter);
+						} catch (IOException ex) {
+							Skript.error("Cannot write to log file '" + logFile + "' (" + logFolder.getPath() + "): " + ExceptionUtils.toString(ex));
 							return;
 						}
 					}
-					w.println("[" + SkriptConfig.formatDate(System.currentTimeMillis()) + "] " + message);
-					w.flush();
+					String levelType;
+					switch (logLevel.intValue()) {
+						case WARNING_LOG:
+							levelType = " WARNING";
+							break;
+						case SEVERE_LOG:
+							levelType = " SEVERE";
+							break;
+						default:
+							levelType = " INFO";
+					}
+					logWriter.println( "[" + SkriptConfig.formatDate(System.currentTimeMillis()) + levelType + "] " + message);
```

Comment:
you can just directly use `logLevel` as the `toString` will return the appropriate name. I am also a little concerned about  the addition of the severity into the file since it's a breaking change. I think when people specify severity it will be specifically for the console log level because that actually has an impact on things. if they wanted it part of their file message, they could just prepend their message. we could disallow the file/severity combo in init :)

Reply:
Well `log` is basically just mimicing the console. It's like sending a message to console but in the form of a log file. so this part "I think when people specify severity it will be specifically for the console log level because that actually has an impact on things." I can see where you are coming from, but that just seems weird in my opinion. "if they wanted it part of their file message, they could just prepend their message. we could disallow the file/severity combo in init" by default this returns `INFO` but I *could* just make it not return INFO. This should really be up for discussion in my honest opinion

Reply:
Yeah I think omitting INFO would be fine, avoids breaking changes

Reply:
im cool with omitting info.

Reply:
it may be useful to allow users to specifically request `info` so the prefix can be used in a consistent manner

Reply:
I'd have to look into this and potentially remove some checks

Reply:
I am just going to omit `info`. Info just seems useless to be honest. I feel like keeping warning and severe is better